tensorflow/python/keras/backend_test.py

```diff
@@ -1736,5 +1736,40 @@ def test_batch_normalization(self):
     self.assertEqual(normed.shape.as_list(), [10, 3, 5, 5])
 
 
+class ControlOpsTests(test.TestCase):
+  @test_util.run_in_graph_and_eager_modes
+  def test_function_switch_basics(self):
+    x = ops.convert_to_tensor(np.array([[1, 2, 3], [4, 5, 6]]),
+                              dtype='float32')
+    y1 = ops.convert_to_tensor(np.array([[1, 2, 3], [4, 5, 6]]),
+                               dtype='float32')
+    x1 = keras.backend.square(x)
+    def true_fun():
+      output = x * x
+      return output
+
+    def false_fun():
+      return y1
+
+    x2 = keras.backend.switch(keras.backend.equal(x, y1), true_fun, false_fun)
+    self.assertAllClose(x2, x1)
+
+    y2 = keras.backend.switch(keras.backend.equal(x, x2), true_fun, false_fun)
+    self.assertAllClose(y2, y1)
```

Comment:
I believe you should be running the graph (e.g. via `backend.function()`) in order to test the behavior of `switch`

Reply:
Updated as per comments